### PR TITLE
Supporting vendor given "shells" configuration file

### PR DESCRIPTION
**Pull request by schubi2**
This PR solves the issue defined in:
https://github.com/shadow-maint/shadow/issues/587

### CODE REVIEW

File: src/chsh.c [hunk 1]
```diff
@@ -129,15 +138,47 @@ static bool is_restricted_shell (const char *sh)
...
 static bool shell_is_listed (const char *sh)
 {
-	char *cp;
 	bool found = false;
 
-#ifndef HAVE_GETUSERSHELL
-	char buf[BUFSIZ];
-	FILE *fp;
-#endif
+#if defined (USE_ECONF) && defined (VENDORDIR)
```
Comment (hallyn) [inline src/chsh.c:143]: Hi,

Could you edit the chsh manpage to give a brief description of the file format?

**Review by hallyn — APPROVED** [review]
I'm not well informed on econf_getKeys(), but this looks good.

**Comment by hallyn** [comment]
Thank you.

**Comment by hallyn** [comment]
Honestly i'd be much happier if the whole shell_is_listed() fn was separately defined for each of the 3 ifdef possibilities.

If you're ok updating the set one more time to do that, that'd be great.  If you prefer not to, let me know and I'll merge this and do it as a separate cleanup.

**Comment by schubi2** [comment]
> Honestly i'd be much happier if the whole shell_is_listed() fn was separately defined for each of the 3 ifdef possibilities.
> 
I see your point. I have cleaned up the code and have defined different complete functions. I hope it is more clear now.

**Review by hallyn — APPROVED** [review]
Looks good to me.  If noone else has any objections, I'll merge it.  (I'll give it a bit of time for others to object)

**Pull request merged**